src/common/hist_util.h

```diff
@@ -116,9 +116,9 @@ struct HistCutMatrix {
 };
 
 /*! \brief Builds the cut matrix on the GPU */
-void DeviceSketch
-  (const SparsePage& batch, const MetaInfo& info,
-   const tree::TrainParam& param, HistCutMatrix* hmat, int gpu_batch_nrows);
+size_t DeviceSketch
```

Comment:
Why are you returning a value from `DeviceSketch()`?

After some thinking, I presume that it is to avoid an extra pass through the data in external memory. If it is so, then this is a good decision. However, it is not obvious from the code, and definitely should be documented.